src/crimson/osd/ops_executer.cc

```diff
@@ -740,95 +741,128 @@ version_t OpsExecuter::get_last_user_version() const
   return pg->get_last_user_version();
 }
 
-void OpsExecuter::make_writeable(std::vector<pg_log_entry_t>& log_entries)
+// a counterpart of the `make_writeable()` in the classical OSD.
+// the main difference is this one executes before, not after,
+// all other operations. the rationale behind the departure is
+// simplicity (no new / old obs, no txn reordering).
+//
+// TODO: make this `static`
```

Comment:
Can you explain why? (same for `prepare_clone`)

Reply:
For the easiness of reasoning. Data dependencies enlisted as method's parameters make them visible from the very beginning. `const` references and `std::as_const()` allows for the read-only semantic at both sides of an interface. `static` guarantees no other parts of the object's state can be used.

This explicitness not always a good thing. Too many / too complex inputs and it hurts. I don't know yet whether `static` will make sense for the cloning part.